Slack.Automation/Promact.Core.Repository/ProjectUserCall/IProjectUserCallRepository.cs

```diff
@@ -106,5 +84,22 @@ public interface IProjectUserCallRepository
         /// <param name="accessToken"></param>
         /// <returns>true or false</returns>
         Task<bool> UserIsAdmin(string userName, string accessToken);
+
+        
+
+        /// <summary>
+        /// Method to call an api from oauth server and get the list of all the projects
+        /// </summary>
+        /// <param name="accessToken"></param>
+        /// <returns>list of all the projects</returns>
+        Task<List<ProjectAc>> GetAllProjects(string accessToken);
+
+        /// <summary>
+        /// Method to call an api from oauth server and get the details of a project using projecId
+        /// </summary>
+        /// <param name="projectId"></param>
+        /// <param name="accessToken"></param>
+        /// <returns>Details of a project</returns>
+        Task<ProjectAc> GetProjectDetails(int projectId, string accessToken);
```

Comment:
Access token is used when we make calls to oauth server. And regarding passing it as a parameter, I followed the same pattern as was being followed earlier and passed it as a parameter in every function. Now after some analysis I found that instead of generating it in the controller and passing it in every function, we can generate it in the repository itself  (Http Client Repository) from where we make request calls to the oauth server and so I have thought of moving its generation process there. Now because of this everyone will have to modify their methods which make calls to oauth server and also the respective test cases. I have created an issue for the same.